t_fault_injection/test_fault_injection_base.py

```diff
@@ -52,26 +54,112 @@ class TestFailFunction(tester.TempestaTest):
+                id="deproxy_h2",
+                msg="cannot establish a new h2 connection",
+                times=-1,
+                should_fail=True,
+                header_len=0,
+                body_len=0,
+                mtu=None,
+                retval=-12,
+            ),
+            param(
+                name="ss_skb_expand_head_tail",
+                func_name="ss_skb_expand_head_tail",
+                id="deproxy_h2",
+                msg="tfw_tls_encrypt: cannot encrypt data",
+                times=1,
+                should_fail=False,
+                header_len=0,
+                body_len=0,
+                mtu=None,
+                retval=-12,
+            ),
+            param(
+                name="ss_skb_to_sgvec_with_new_pages",
+                func_name="ss_skb_to_sgvec_with_new_pages",
+                id="deproxy_h2",
+                msg="tfw_tls_encrypt: cannot encrypt data",
+                times=1,
+                should_fail=False,
+                header_len=0,
+                body_len=0,
+                mtu=None,
+                retval=-12,
+            ),
+            param(
+                name="ss_skb_to_sgvec_with_new_pages_long_resp",
+                func_name="ss_skb_to_sgvec_with_new_pages",
+                id="deproxy_h2",
+                msg="tfw_tls_encrypt: cannot encrypt data",
+                times=1,
+                should_fail=False,
+                header_len=50000,
+                body_len=100000,
+                mtu=100,
+                retval=-12,
+            ),
         ]
     )
     @dmesg.unlimited_rate_on_tempesta_node
-    def test(self, name, id, msg, retval):
+    def test(self, name, func_name, id, msg, times, should_fail, header_len, body_len, mtu, retval):
+        if mtu:
+            try:
+                dev = sysnet.route_dst_ip(remote.client, tf_cfg.cfg.get("Tempesta", "ip"))
+                prev_mtu = sysnet.change_mtu(remote.client, dev, mtu)
+                self._test(
+                    name, func_name, id, msg, times, should_fail, header_len, body_len, retval
+                )
+            finally:
+                sysnet.change_mtu(remote.client, dev, prev_mtu)
+        else:
+            self._test(name, func_name, id, msg, times, should_fail, header_len, body_len, retval)
+
+    def _test(self, name, func_name, id, msg, times, should_fail, header_len, body_len, retval):
         """
         Basic test to check how Tempesta FW works when some internal
         function fails. Function should be marked as ALLOW_ERROR_INJECTION
         in Tempesta FW source code.
         """
         self.start_all_services(client=False)
 
-        # Write function name to special debug fs file. 
-        cmd = f"echo {name} > /sys/kernel/debug/fail_function/inject"
+        if header_len != 0 or body_len != 0:
+            server = self.get_server("deproxy")
+            self.set_response(server, header_len, body_len)
```

Comment:
I think you should use a completed response here. For example:

```python3

param(
    name="ss_skb_to_sgvec_with_new_pages_long_resp",
    func_name="ss_skb_to_sgvec_with_new_pages",
    id="deproxy_h2",
    response=create_http_response(header_len=1000, body_len=5000),
)
def test(...):
    self.get_server("deproxy").set_response(response)
```

I believe this will make the test cleaner.